main.go

```diff
@@ -63,24 +192,80 @@ var httpClient = http.Client{
 }
 
 type exporterOpts struct {
-	autoDiscover bool
-	interval     time.Duration
-	localURL     string
-	masterURL    string
+	autoDiscoverInterval time.Duration
+	interval             time.Duration
+	mode                 string
+	queryURL             string
+}
+
+type exporterTaskInfo struct {
+	FrameworkName string
+	TaskName      string
+}
+
+type framework struct {
+	Id            string
+	Name          string
+	UsedResources *resources `json:"used_resources"`
 }
 
+type masterState struct {
+	Frameworks []*framework
+	Hostname   string
+	Slaves     []*slave
+}
+
+type snapshotMetric struct {
+	desc        *prometheus.Desc
+	snapshotKey string
+	valueType   prometheus.ValueType
+}
+
+type metricsSnapshot map[string]float64
+
 type periodicExporter struct {
 	sync.RWMutex
-	errors    *prometheus.CounterVec
-	masterURL *url.URL
-	metrics   []prometheus.Metric
-	opts      *exporterOpts
-	slaves    struct {
+	errors   *prometheus.CounterVec
+	metrics  []prometheus.Metric
+	opts     *exporterOpts
+	queryURL *url.URL
+	slaves   struct {
 		sync.Mutex
 		urls []string
 	}
 }
 
+type resources struct {
+	Cpus float64
```

Comment:
`Cpus` -> `CPUs`, same for `Pid` -> `PID`, `Id` -> `ID`, `FrameworkId` -> `FrameworkID`.